src/app/services/utils.service.ts

```diff
@@ -178,23 +170,45 @@ export class UtilsService {
         }).format(date);
 
       default:
-        return new Intl.DateTimeFormat('en-GB', {
-          hour12: true,
-          hour: 'numeric',
-          minute: 'numeric',
-          month: 'short',
-          day: 'numeric',
-          year: 'numeric'
-        }).format(date);
+      return this.dateFormatter(date) + ' ' + new Intl.DateTimeFormat('en-GB', {
+        hour12: true,
+        hour: 'numeric',
+        minute: 'numeric'
+      }).format(date);
+    }
+  }
+
+  dateFormatter(date: Date) {
+    const today = new Date();
+    let formattedDate = new Intl.DateTimeFormat('en-GB', {
+      month: 'short',
+      day: 'numeric',
+      year: 'numeric'
+    }).format(date);
+
+    if (date.getFullYear() === today.getFullYear() && date.getMonth() === today.getMonth()) {
+      if (date.getDate() === today.getDate() - 1) {
+        formattedDate = 'Yesterday';
+      }
+      if (date.getDate() === today.getDate()) {
+        formattedDate = 'Today';
+      }
+      if (date.getDate() === today.getDate() + 1) {
+        formattedDate = 'Tomorrow';
+      }
     }
+    return formattedDate;
   }
 
-  timeComparer(timeString: string, comparedString?: string) {
+  timeComparer(timeString: string, comparedString?: string, compareDate?: boolean) {
     const time = new Date(timeString + 'Z');
     let compared = new Date();
     if (comparedString) {
       compared = new Date(comparedString + 'Z');
     }
+    if (compareDate && (time.getDate() === compared.getDate() && time.getTime() >= compared.getTime())) {
```

Comment:
`time.getTime() >= compared.getTime()` is not necessary.
Also, we need to compare year and month here to make sure it's the same day